src/OrbitGl/CaptureWindowTest.cpp

```diff
@@ -161,113 +174,193 @@ class NavigationTestCaptureWindow : public CaptureWindow, public testing::Test {
   CaptureClientAppInterfaceFake capture_client_app_fake_;
   std::unique_ptr<orbit_client_data::CaptureData> capture_data_;
 
-  void CreateSchedulerTimers() {
-    using orbit_client_protos::TimerInfo;
-    TimerInfo timer;
-    timer.set_start(0);
-    timer.set_end(100);
-    timer.set_thread_id(TrackTestData::kThreadId);
-    timer.set_processor(0);
-    timer.set_depth(0);
-    timer.set_type(TimerInfo::kCoreActivity);
-    time_graph_->ProcessTimer(timer, nullptr);
+  void AddTimers() {
+    auto timers = TrackTestData::GenerateTimers();
+    for (auto& timer : timers) {
+      time_graph_->ProcessTimer(timer, nullptr);
+    }
   }
 };
 
-TEST_F(NavigationTestCaptureWindow, ZoomTimeWorksAsExpected) {
+TEST_F(NavigationTestCaptureWindow, ZoomTimeWorksInTheMiddleOfTimeGraph) {
+  PreRender();
   ExpectInitialState();
-
   const Vec2i kTimeGraphSize = viewport_.WorldToScreen(time_graph_->GetSize());
 
-  // Zoom time: Zoom in and out horizontally, assure we've zoomed in, and zoomed out
-  // again
-  // Test 1: Middle of the capture window
   int x = kTimeGraphSize[0] / 2;
-  int y = kTimeGraphSize[1] - 5;
+  int y = kTimeGraphSize[1] - kBottomSafetyMargin;
+
   MouseWheelMoved(x, y, 1, false);
-  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kInBetween);
+  PreRender();
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kMiddle);
+
   MouseWheelMoved(x, y, -1, false);
+  PreRender();
   ExpectInitialState();
 
   // Keyboard
   MouseMoved(x, y, false, false, false);
   KeyPressed('W', false, false, false);
-  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kInBetween);
+  PreRender();
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kMiddle);
+
   KeyPressed('S', false, false, false);
+  PreRender();
   ExpectInitialState();
+}
+
+TEST_F(NavigationTestCaptureWindow, ZoomTimeWorksAtRandomPositions) {
+  PreRender();
+  ExpectInitialState();
+  const Vec2i kTimeGraphSize = viewport_.WorldToScreen(time_graph_->GetSize());
+
+  std::random_device rd;
+  std::mt19937 mt(rd());
+  std::uniform_int_distribution<int> dist(0, kTimeGraphSize[0]);
+
+  constexpr int kNumTries = 100;
+  for (int i = 0; i < kNumTries; ++i) {
+    int x = dist(mt);
+    int y = kTimeGraphSize[1] - kBottomSafetyMargin;
+
+    // Zoom in twice, then zoom out twice. Check that the intermediate states match
+    MouseWheelMoved(x, y, 1, false);
+    PreRender();
+    ExpectIsHorizontallyZoomedIn(PosWithinCapture::kAnywhere);
+
+    float last_slider_pos = slider_->GetPosRatio();
+    double last_min_time = time_graph_->GetMinTimeUs();
+    double last_max_time = time_graph_->GetMaxTimeUs();
+    MouseWheelMoved(x, y, 1, false);
+    PreRender();
+    ExpectIsHorizontallyZoomedIn(PosWithinCapture::kAnywhere);
+
+    MouseWheelMoved(x, y, -1, false);
+    PreRender();
+    EXPECT_NEAR(slider_->GetPosRatio(), last_slider_pos, kSliderPosEpsilon);
```

Comment:
I expected to have issues like this. Nice catch Florian! :)

This should be solved in the future and replace with EXPECT_DOUBLE_EQ